DirectProgramming/DPC++/StructuredGrids/1d_HeatTransfer/src/1d_HeatTransfer.cpp

```diff
@@ -60,114 +60,177 @@ void Usage(string programName) {
   cout << " i : Number of timesteps \n";
 }
 
-//************************************
-// Function description: initialize the array.
-//************************************
-void Initialize(float* array, unsigned int num, unsigned int idx) {
-  for (unsigned int i = idx; i < num; i++) array[i] = 0.0f;
+//
+// Initialize the temperature arrays
+//
+void Initialize(float* arr, float* arr_next, unsigned int num) {
+  arr[0] = arr_next[0] = initial_temperature;
```

Comment:
This looks wrong, the arr[0]=.... line should be last to not be overwritten by the loop.  Or loop start from 1.

Reply:
Good catch. Initializing the array to all 0 masked my other bugs.